src/state-client.ts

```diff
@@ -284,101 +323,144 @@ class StateClient {
+  public multiGetExtensionViewerState = <ResponseType = unknown>(identifier: string, users: string[]) =>
+    this.signedRequest<String, ResponseType>(identifier, 'GET', `extension_viewer_state?user_ids=${users.join(',')}`);
 
   /** @ignore */
-  public multiGetExtensionViewerState = (identifier: string, users: string[]) =>
-    this.signedRequest(identifier, 'GET', `extension_viewer_state?user_ids=${users.join(',')}`);
+  public setExtensionSecretState = <DataType = unknown, ResponseType = unknown>(identifier: string, state: DataType) =>
+    this.postState<DataType, ResponseType>(identifier, ServerState.EXTENSION_SECRET, state);
 
   /** @ignore */
-  public setExtensionSecretState = (identifier: string, state: any) =>
-    this.postState(identifier, ServerState.EXTENSION_SECRET, JSON.stringify(state));
+  public setChannelState = <DataType = unknown, ResponseType = unknown>(identifier: string, state: DataType) =>
+    this.postState<DataType, ResponseType>(identifier, ServerState.CHANNEL, state);
 
   /** @ignore */
-  public setChannelState = (identifier: string, state: any) =>
-    this.postState(identifier, ServerState.CHANNEL, JSON.stringify(state));
+  public setChannelConfig = <DataType = unknown, ResponseType = unknown>(identifier: string, config: DataType) =>
+    this.postConfig<DataType, ResponseType>(identifier, ServerConfig.CHANNEL, config);
 
   /** @ignore */
-  public setExtensionState = (identifier: string, state: any) =>
-    this.postState(identifier, ServerState.EXTENSION, JSON.stringify(state));
+  public setExtensionConfig = <DataType = unknown, ResponseType = unknown>(identifier: string, config: DataType) =>
+    this.postConfig<DataType, ResponseType>(identifier, ServerConfig.EXTENSION, config);
+
+  /** @ignore */
+  public patchChannelConfig = (identifier: string, multiConfig: any) =>
+    this.signedRequest(identifier, 'PATCH', ServerConfig.CHANNEL, JSON.stringify(multiConfig));
+
+  /** @ignore */
+  public patchExtensionConfig = (identifier: string, multiConfig: any) =>
+    this.signedRequest(identifier, 'PATCH', ServerConfig.EXTENSION, JSON.stringify(multiConfig));
+
+  /** @ignore */
+  public setExtensionState = <DataType = unknown, ResponseType = unknown>(identifier: string, state: DataType) =>
+    this.postState<DataType, ResponseType>(identifier, ServerState.EXTENSION, state);
 
   /** @ignore */
   public getAccumulation = (identifier: string, id: string, start: number) =>
-    this.signedRequest(identifier, 'GET', `accumulate?id=${id || 'default'}&start=${start}`);
+    this.signedRequest<Number, AccumulateData>(identifier, 'GET', `accumulate?id=${id || 'default'}&start=${start}`);
 
   /** @ignore */
   public accumulate = (identifier: string, id: string, data: any) =>
-    this.signedRequest(identifier, 'POST', `accumulate?id=${id || 'default'}`, JSON.stringify(data));
+    this.signedRequest<String, Object>(identifier, 'POST', `accumulate?id=${id || 'default'}`, JSON.stringify(data));
```

Comment:
`data` as `any`. Needs a type generic. Also if you pass `JSON.stringify(data)` as the `data` parameter, and the `signedRequest` does a `JSON.stringify` on the string, do you get a doubly-encoded string? Does `signedRequest` need to check the type of `data` and only stringify non-strings?